apps/passport-client/src/participant.ts

```diff
@@ -20,8 +20,20 @@ export async function pollParticipant(
       console.log("[USER_POLL] Participant not found, skipping update");
       return;
     }
+
     const participant = await response.json();
-    dispatch({ type: "set-self", self: participant });
+
+    try {
+      await dispatch({ type: "set-self", self: participant });
+    } catch (e: any) {
+      console.log("failed to set self");
+      if (typeof e.message === "string") {
+        const msgString: string = e.message;
+        if (msgString.includes("mismatch")) {
+          dispatch({ type: "participant-invalid" });
```

Comment:
cchecking the error message string is a bit fragile.. i recommend just calling the participant invalid function from inside the `set-self` handler instead of throwing

Reply:
OK 👍